File: demos/crewai-agent/src/agent.py
from crewai import Agent

from tools.calculator import calculator
from tools.weather import get_weather
from tools.file_reader import read_file
from tools.search import web_search

tools = [calculator, get_weather, read_file, web_search]

assistant = Agent(
    role="Helpful Assistant",
    goal="Answer user questions accurately using the available tools",
    backstory=(
        "You are a helpful assistant with access to a calculator, weather lookup, "
        "file reader, and web search. You MUST use the appropriate tool to answer "
        "questions — never guess or make up information."
    ),
    tools=tools,
    verbose=False,
)

File: demos/langchain-tool-agent/src/agent.ts
import { createAgent } from "langchain";
import { calculator } from "./tools/calculator.js";
import { getWeather } from "./tools/weather.js";
import { readFile } from "./tools/fileReader.js";
import { webSearch } from "./tools/search.js";

export const tools = [calculator, getWeather, readFile, webSearch];

export const agent = createAgent({
  model: "openai:gpt-5.4-nano",
  tools,
  systemPrompt: `You are a helpful assistant with access to the following tools:
- calculator: for arithmetic operations
- get_weather: for checking weather in cities
- read_file: for reading local files
- web_search: for searching the web

Use the appropriate tool(s) to answer the user's questions. Be concise in your responses.`,
});
